tests/integ/conftest.py

```diff
@@ -252,3 +293,170 @@ def temp_schema(connection, session, local_testing_mode) -> None:
             )
             yield temp_schema_name
             cursor.execute(f"DROP SCHEMA IF EXISTS {temp_schema_name}")
+
+
+class TracebackHistory(QueryListener):
+    """
+    Helper class to track queries, tracebacks and request IDs to merge them with REST requests tracked
+    through VCRpy.
+    """
+
+    def __init__(self) -> None:
+        self.queries = []
+        self.tracebacks = []
+        self.request_ids = []
+
+    def _notify(self, query_record: QueryRecord, *args, **kwargs) -> None:
+
+        formatted_lines = traceback.format_stack()
+
+        # exclude all the wrapper code, i.e. start with files in snowflake/snowpark (or tests)
+        idx = next(
+            i
+            for i, line in enumerate(formatted_lines)
+            if "snowflake/snowpark" in line or "tests/" in line
+        )
+        formatted_lines = formatted_lines[idx:]
+
+        # remove traceback related to _internal/server_connection.py
+        idx = next(
+            i
+            for i, line in enumerate(formatted_lines)
+            if "_internal/server_connection.py" in line
+        )
+
+        # cleanup traceback from telemetry, because there's no added value in displaying it.
+        formatted_lines = formatted_lines[
+            : idx - 1
+        ]  # -1 to also exclude the call of session._conn.execute
+
+        formatted_lines = [
+            line for line in formatted_lines if "telemetry.py" not in line
+        ]
+
+        request_id = kwargs["requestId"]
+
+        self.request_ids.append(request_id)
+        self.queries.append(query_record)
+        self.tracebacks.append(formatted_lines)
+
+
+# Disable phase0 Snowpark IR check by setting autouse=False,
+# by default enabled for all tests (except modin/) to check that Snowpark python requests contain
+# the dataframeAST field holding the encoded representation.
```

Comment:
dataframeAst. I think the standard terminology in JSON is "property" instead of "field".

Reply:
let me update the comment in that case!